source/utilities.cc

```diff
@@ -1812,43 +1904,65 @@ namespace aspect
                                          " in the data file " + filename + ". Please remove duplication to "
                                          "allow for unique association between column and name."));
 
-                  data_component_names.push_back(column_name_or_data);
+                  column_names.push_back(column_name_or_data);
                 }
               ++name_column_index;
             }
         }
 
-      /**
-       * Create table for the data. This peculiar reinit is necessary, because
-       * there is no constructor for Table, which takes TableIndices as
-       * argument.
-       */
-      data.resize(components);
-      maximum_component_value.resize(components,-std::numeric_limits<double>::max());
+      // Create table for the data. This peculiar reinit is necessary, because
+      // there is no constructor for Table, which takes TableIndices as
+      // argument.
       Table<dim,double> data_table;
       data_table.TableBase<dim,double>::reinit(table_points);
-      std::vector<Table<dim,double> > data_tables(components+dim,data_table);
+      std::vector<Table<dim,double> > data_tables(components, data_table);
+
+      std::vector<std::vector<double>> coordinate_values(dim);
+      for (unsigned int d=0; d<dim; ++d)
+        coordinate_values[d].resize(table_points[d]);
 
+      if (column_names.size()==0)
+        {
+          // set default column names:
+          for (unsigned int c=0; c<components; ++c)
+            column_names.push_back("column " + Utilities::int_to_string(c,2));
+        }
 
-      // Read data lines
+      // Finaylly read data lines
       unsigned int read_data_entries = 0;
       do
         {
+          // what row and column of the file are we in?
           const unsigned int column_num = read_data_entries%(components+dim);
+          const unsigned int row_num = read_data_entries/(components+dim);
+          TableIndices<dim> idx = compute_table_indices(table_points, row_num);
 
-          if (column_num >= dim)
+          if (column_num < dim)
             {
-              temp_data *= scale_factor;
-              maximum_component_value[column_num-dim] = std::max(maximum_component_value[column_num-dim], temp_data);
-            }
+              // This is a coordinate. Store (and check that they are consistent)
+              const double old_value = coordinate_values[column_num][idx[column_num]];
 
-          data_tables[column_num](compute_table_indices(read_data_entries)) = temp_data;
+              AssertThrow(old_value == 0. ||
+                          (std::abs(old_value-temp_data) < 1e-8*std::abs(old_value)),
+                          ExcMessage("Invalid coordinate "
+                                     + Utilities::int_to_string(column_num) + " in row "
+                                     + Utilities::int_to_string(row_num)
+                                     + " in file " + filename));
```

Comment:
Can you clarify in this error message that the problem is that the new position has moved compared to the old position?